transport-native-unix-common-tests/src/main/java/io/netty/channel/unix/tests/UnixChannelUtilTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.channel.unix.tests;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.buffer.CompositeByteBuf;
+import io.netty.buffer.UnpooledByteBufAllocator;
+import org.junit.Test;
+
+import java.util.Collections;
+import java.util.LinkedList;
+import java.util.List;
+
+import static io.netty.channel.unix.Limits.IOV_MAX;
+import static io.netty.channel.unix.UnixChannelUtil.isBufferCopyNeededForWrite;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class UnixChannelUtilTest {
+    @Test
+    public void testIsBufferCopyNeededForWrite() {
+        ByteBufAllocator alloc = new UnpooledByteBufAllocator(false);
```

Comment:
can we move this test content into a private method and pass in the standard allocators?

```java
@Test
public void testPooledAllocatorIsBufferCopyNeededForWrite() {
  testIsBufferCopyNeededForWrite(PooledByteBufAllocator.DEFAULT);
}

@Test
public void testUnPooledAllocatorIsBufferCopyNeededForWrite() {
  testIsBufferCopyNeededForWrite(UnPooledByteBufAllocator.DEFAULT);
}

private void testIsBufferCopyNeededForWrite(ByteBufAllocator alloc) {
...
}